### ISSUE TITLE
BT-701: Notice Identifier: to be assigned by initiating system

### ISSUE THREAD

**Issue by rantati7** [eForms/eForms#6]
Currently, the Notice Identifier is generated and assigned in TED. However this complicates matters significantly as we have to wait for the notice identifier to come from TED and also at times we have to go back to TED to check this identifier. 

**A solution:** 

Similar to procudure identifier this identifier should be unique and generated already in the initiating platform. This is not purely an IT issue as the syntax should be agreed, 

Could this number be combined with the procedure identifier? I mean could the procedure identifier have addtional e..g two or x numbers that also identifyu the notice? 

Is there a syntax in the OCDS that we could use?

**Comment by JachymHercher** [comment]
For the discussions on semantic / asemantic identifiers, please see #5. 

The problem with adding a few characters to a procedure ID and taking it as a notice ID are PINs only - they can be about several procedures. At that moment, the approach stops working, because a notice would need two different identifiers. For that reason, same as for the procedure identifier, we are slightly inclined towards having an asemantic identifier. 

We would be glad for more views on this.

**Comment by tinne** [comment]
Instead of creating another ID creation and distribution system as discussed in #5, would it not be sufficient to generate a notice identifier that is unique within the procedure (or any relevant procedure)?

This way, irrespective whether the identifier system is asemantic or semantic, the pair (Procedure Identifier, Notice Identifier) would define a single notice. In case of the need for a canonical identifier (and you cannot implement #55), define any total order in the Procedure Identifier and use the pair (smallest Procedure Identifier, Notice Identifier) as the canonical one.

**Comment by JachymHercher** [comment]
Yes, it is sufficient that the notice ID is unique within the relevant procedure / all the relevant procedures (for PINs only, possibly).

Assuming that PINs can refer to only one procedure in line with #55, what we still need to analyze is whether this notice ID could be a sequential number, where the first notice gets "-1", second "-2" etc. Would this cause problems when TED rejects notices, esp. in cases where several notices (e.g. changes) might be sent to TED within a few days, before the notice is published?

If the ordinal approach is not possible, then I would assume the notice identifier be just 1-2 random characters.

**Comment by tinne** [comment]
The question is whether it poses a problem to anyone if the line of finally published notices is broken.

Say, the originating person or system generates these suffixes (I assume you don't mean negative numbers by ”-1“...) in ascending order: One would be guaranteed that the published notices have strictly ascending numbers when some intermediate notices are rejected or withdrawn.

This could hopefully replace both `no_doc_ext` and `submission_id`. In case anyone needs a continuous sequence, this can be generated and assigned on publication, i.e., the moment `no_doc_ojs` is produced.

Also, this would remove the versioning issue raised in #71/#72.

**Comment by paulakeen** [comment]
# Conclusion

If issue [#5]([url](https://github.com/eForms/eForms/issues/5) is not solved, this [issue #6](https://github.com/eForms/eForms/issues/6) could not be solved either.
